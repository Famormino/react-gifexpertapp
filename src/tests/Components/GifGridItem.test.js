import React from 'react';
import { shallow } from "enzyme"
import { GifGridItem } from "../../Components/GifGridItem"

describe('Pruebas en <GifGridItem />', () => {

    const title = 'Un título';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem title = { title } url = { url }  /> )

    test('Debe de mostrar el texto correctamente', () => {
    
        expect( wrapper ).toMatchSnapshot()
    });

    test('Debe tener un párrafo con el title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );

    });

    test('Debe tener la img igual al url y alt de los props', () => {

        const img = wrapper.find('img');
       //console.log(img.prop('src'))
       expect( img.prop('src') ).toBe( url );
       expect( img.prop('alt') ).toBe( title );
        
    });

    test('debe tener animate__fadeIn', () => {

        const div = wrapper.find('div');
        const className = div.prop('className')

        expect( className.includes('animate__fadeIn') ).toBe( true );
        
    })
    
    
    
    
})
