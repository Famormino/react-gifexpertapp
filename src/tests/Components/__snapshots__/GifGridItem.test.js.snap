// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe de mostrar el texto correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p
    className="title-category"
  >
     
    Un título
     
  </p>
</div>
`;
